libs/SalesforceHybrid/src/com/salesforce/androidsdk/phonegap/plugin/SalesforceNetworkPlugin.java

```diff
@@ -244,7 +244,7 @@ private RestClient getRestClient(boolean doesNotRequireAuth) {
             return null;
         }
         if (doesNotRequireAuth) {
-            return SalesforceHybridSDKManager.getInstance().getClientManager().peekUnauthenticatedRestClient();
+            return currentActivity.buildClientManager().peekUnauthenticatedRestClient();
```

Comment:
Tests use an activity that override buildClientManager() to build a ClientManager using test credentials.